src/android/com/microsoftopentechnologies/intellij/forms/CreateQueueForm.java

```diff
@@ -0,0 +1,169 @@
+            public void removeUpdate(DocumentEvent documentEvent) {
+                changedName();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent documentEvent) {
+                changedName();
+            }
+        });
+
+        contentPane.registerKeyboardAction(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                onCancel();
+            }
+        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
+    }
+
+    private void changedName() {
+        buttonOK.setEnabled(nameTextField.getText().length() > 0);
+    }
+
+    private void onOK() {
+        final String name = nameTextField.getText();
+        if (name.length() < 3 || name.length() > 63 || !name.matches("^[a-z0-9](?!.*--)[a-z0-9-]+[a-z0-9]$")) {
+            JOptionPane.showMessageDialog(this, "Queue names must start with a letter or number, and can contain only letters, numbers, and the dash (-) character.\n" +
+                    "Every dash (-) character must be immediately preceded and followed by a letter or number; consecutive dashes are not permitted in container names.\n" +
+                    "All letters in a container name must be lowercase.\n" +
+                    "Queue names must be from 3 through 63 characters long.", "Service Explorer", JOptionPane.ERROR_MESSAGE);
+            return;
+        }
+
+        ProgressManager.getInstance().run(new Task.Backgroundable(project, "Creating queue...", false) {
+
+            @Override
+            public void run(@NotNull ProgressIndicator progressIndicator) {
+                try {
+                    progressIndicator.setIndeterminate(true);
+
+
+                    for (Queue queue : AzureSDKManagerImpl.getManager().getQueues(storageAccount)) {
+
+                        if (queue.getName().equals(name)) {
+                            ApplicationManager.getApplication().invokeLater(new Runnable() {
+                                @Override
+                                public void run() {
+                                    JOptionPane.showMessageDialog(null, "A queue with the specified name already exists.", "Service Explorer", JOptionPane.ERROR_MESSAGE);
+                                }
+                            });
+
+                            return;
+
+                        }
+                    }
+
+
+                    Queue queue = new Queue(name, "", 0, storageAccount.getSubscriptionId());
+                    AzureSDKManagerImpl.getManager().createQueue(storageAccount, queue);
+
+                    if (onCreate != null) {
+                        ApplicationManager.getApplication().invokeLater(onCreate);
+                    }
+                } catch(AzureCmdException e) {
+                    UIHelper.showException("Error creating queue", e, "Service explorer", false, true);
+                }
+            }
+        });
+
+        dispose();
```

Comment:
Is there a potential problem here in case the form gets disposed before the task started on line 115 can complete?